DataFormats/NanoAOD/interface/FlatTable.h

```diff
@@ -0,0 +1,159 @@
+#ifndef DataFormats_NanoAOD_FlatTable_h
+#define DataFormats_NanoAOD_FlatTable_h
+
+#include <cstdint>
+#include <vector>
+#include <string>
+#include <boost/range/sub_range.hpp>
+#include <FWCore/Utilities/interface/Exception.h>
+#include <DataFormats/PatCandidates/interface/libminifloat.h>
+
+namespace flatTableHelper {
+    template<typename T> struct MaybeMantissaReduce { 
+        MaybeMantissaReduce(int mantissaBits) {}
+        inline T one(const T &val) const  { return val; }
+        inline void bulk(boost::sub_range<std::vector<T>> data) const {  }
+    };
+    template<> struct MaybeMantissaReduce<float> {
+        int bits_; 
+        MaybeMantissaReduce(int mantissaBits) : bits_(mantissaBits) {}
+        inline float one(const float &val) const  { return (bits_ > 0 ? MiniFloatConverter::reduceMantissaToNbitsRounding(val, bits_) : val); }
+        inline void bulk(boost::sub_range<std::vector<float>> data) const { if (bits_ > 0) MiniFloatConverter::reduceMantissaToNbitsRounding(bits_, data.begin(), data.end(), data.begin()); }
```

Comment:
pass by const ref?

Reply:
sub_range is an iterator pair, it's to be passed by value